libraries/chain/include/eosio/chain/wasm_interface_private.hpp

```diff
@@ -158,6 +162,10 @@ namespace eosio { namespace chain {
          return it->module;
       }
 
+      bool should_always_oc_tierup()const {
+         return wasm_runtime_time == wasm_interface::vm_type::eos_vm_oc || eosvmoc_tierup == wasm_interface::vm_oc_enable::oc_all;
```

Comment:
I don't think `wasm_runtime_time == wasm_interface::vm_type::eos_vm_oc` should be in here. `wasm_runtime_time == wasm_interface::vm_type::eos_vm_oc` is meant to be "OC in _non_-tier up mode" and all the code back here
https://github.com/AntelopeIO/leap/blob/6008c728927f524e06a6d77463c5a5c165190c18/libraries/chain/wasm_interface.cpp#L93
should be skipped entirely in non-tier up mode. It's not clear what the ramifications are for having both a synchronous and asynchronous code_cache at the same time (which is what I think might be happening in this case.. not sure.. regardless it def wasn't originally intended to operate this way)

I think the line above should be something more on the order of
```c++
if(my->eosvmoc &&
  (my->eosvmoc_tierup == wasm_interface::vm_oc_enable::oc_all || context.should_use_eos_vm_oc())) { 
```

Reply:
Thanks. I missed that `eos_vm_oc` didn't flow throw here.

Reply:
It seems a non issue from a functional standpoint -- I remember now that `my->eosvmoc` isn't created unless in `oc_auto` or `oc_all` mode,
https://github.com/AntelopeIO/leap/blob/a8f20cebf41e5b0706616d9defd13e5de95a3027/libraries/chain/include/eosio/chain/wasm_interface_private.hpp#L93-L96
so nothing wonky would have happened since `my->eosvmoc &&` would have always failed out. But yeah def better now with the tweak.